cellmlmanip/model.py

```diff
@@ -802,8 +803,13 @@ def convert_variable(self, original_variable, units, direction):
         if cf == 1:
             # No conversion necessary. The method above will ensure a factor close to 1 is returned as 1.
             return original_variable
-        # Make the conversion factor a number symbol with explicit units
-        cf = self.create_quantity(cf, units / original_variable.units)
+
+        if isinstance(cf, numbers.Number):
+            # Make the conversion factor a number symbol with explicit units
+            cf = self.create_quantity(cf, units / original_variable.units)
+        elif isinstance(cf, sympy.mul.Mul) and 1.0 in cf.args:
```

Comment:
Come to think of it, it'd make much more sense for this check to happen **inside** `get_conversion_factor`, just like we check for a factor close to 1.0 there.

Reply:
good point